src/sign-in-xplat/MainPage.xaml.cs

```diff
@@ -48,36 +53,41 @@ private async void OnSignInClicked(object sender, EventArgs e)
-                            s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)
-                                                              .ExecuteAsync();
-                        }
-                        catch (MsalUiRequiredException ex)
-                        {
-                            // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenAsync to acquire a token
-                            Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
-
-                            s_authResult = await s_publicClientApp.AcquireTokenInteractive(s_scopes)
-#if ANDROID
-                                                              .WithParentActivityOrWindow(Microsoft.Maui.Essentials.Platform.CurrentActivity)
-#endif
-                                                              .ExecuteAsync()
-                                                              .ConfigureAwait(false);
-                        }
+                        // Signs in the user and obtains an Access token for MS Graph
+                        s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)
+                                                            .ExecuteAsync();
+                    }
+                    catch (MsalUiRequiredException ex)
+                    {
+                        // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenAsync to acquire a token
+                        Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
 
-                        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", s_authResult.AccessToken);
-                    }));
+                        s_authResult = await s_publicClientApp.AcquireTokenInteractive(s_scopes)
+                                                            .ExecuteAsync()
+                                                            .ConfigureAwait(false);
+                    }
+                    
+                    return new AuthenticationHeaderValue("bearer", s_authResult.AccessToken);
+                }
 
                 // Call the /me endpoint of Graph
-                User graphUser = await graphClient.Me.Request().GetAsync();
+                User graphUser;
+#if ANDROID
+                s_httpClient.DefaultRequestHeaders.Authorization = await SignInUserAndGetTokenUsingMSAL();
+                graphUser = await s_httpClient.GetFromJsonAsync<User>($"{s_graphURL}/me");
+#else
+                GraphServiceClient graphClient = new GraphServiceClient(s_graphURL,
+                    new DelegateAuthenticationProvider(
+                        async (requestMessage) => requestMessage.Headers.Authorization = await SignInUserAndGetTokenUsingMSAL()));
+                graphUser = await graphClient.Me.Request().GetAsync();
+#endif

```

Comment:
Let's just use the http-based on for windows and Android and remove the Graph SDK completely.  We don't need to make this overly complicated :)  That way when I test this next, I'll test on both android and windows to make sure it's functioning with the same code path.

Reply:
you got it @ckittel 

done | addressed from [d3a76](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/17/commits/d3a76701e251d2667963144cd379f11ecb34163a)

Reply:
Don't forget to remove the nuget package reference as well.

Reply:
Oh, you're still pulling it into the Graph `User` object.  Nevermind my comment above then.  Marking resolved.